src/Interpreters/InterpreterInsertQuery.cpp

```diff
@@ -149,6 +149,24 @@ Block InterpreterInsertQuery::getSampleBlock(
     return getSampleBlock(names, table, metadata_snapshot);
 }
 
+std::optional<Names> InterpreterInsertQuery::getInsertColumnNames() const
+{
+    auto const * insert_query = query_ptr->as<ASTInsertQuery>();
+    if (!insert_query || !insert_query->columns)
+        return std::nullopt;
+
+    auto table = DatabaseCatalog::instance().getTable(getDatabaseTable(), getContext());
+    Names names;
```

Comment:
reserve

Reply:
Copied similar code from `getSampleBlock` and didn't noticed it, good catch, will add it to both places